docs/performance-benchmarking.md

```diff
@@ -0,0 +1,225 @@
+# Performance Benchmarking
+
+This document describes how performance benchmarking works in the OpenTelemetry Collector project, including how to write benchmarks, run them locally, and how performance regressions are detected in CI.
+
+## Overview
+
+The OpenTelemetry Collector uses a combination of Go's built-in benchmarking framework and [CodSpeed](https://codspeed.io) for continuous performance monitoring. This infrastructure helps us detect performance regressions early and track performance improvements over time.
+
+## Running Benchmarks Locally
+
+### Running All Benchmarks
+
+To run all benchmarks across all modules:
+
+```shell
+make gobenchmark
+```
+
+This will:
+- Run benchmarks in all modules
+- Set the `MEMBENCH` environment variable to include memory benchmarks
+- Generate a `benchmarks.txt` file with results
+
+### Running Time-Only Benchmarks
+
+For faster iteration with time-based benchmarks only:
```

Comment:
The use case here wouldn't really be to run faster, since the benchmarks that are excluded are actually the fastest ones.


```suggestion
Some benchmarks run too quickly to provide meaningful duration results. To skip them:
```